src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1001,6 +1003,43 @@ private void InitODataParser()
             PopulateColumnDefinitionWithHasDefault(
                 sourceDefinition,
                 columnsInTable);
+
+            if (GetDatabaseType() is not DatabaseType.MySQL && entity is not null && entity.Source.Type is EntitySourceType.Table)
```

Comment:
Is this restricted only for tables? Cant we infer this for views?

Reply:
No, for views the metadata is not populated correctly.

Reply:
For a view like this:

```tsql
-- Create base table
CREATE TABLE Employees (
    EmployeeID INT PRIMARY KEY,
    FirstName NVARCHAR(50),
    LastName NVARCHAR(50)
);

-- Insert sample data
INSERT INTO Employees (EmployeeID, FirstName, LastName)
VALUES (1, 'John', 'Doe'),
       (2, 'Jane', 'Smith');

-- Create view
CREATE VIEW EmployeeView AS
SELECT
    EmployeeID,
    FirstName,
    LastName,
    CONCAT(FirstName, ' ', LastName) AS FullName
FROM
    Employees;
```

Reply:
Where you would expect EmployeeView.FullName to be a computed column, the metadata comes out to be as:
<img width="738" alt="image" src="https://github.com/Azure/data-api-builder/assets/34566234/52e7361f-374d-41f2-b3b6-7594625b3156">